### PR TITLE
MODCON-25 user retrieving fix

### PR DESCRIPTION
**Pull request by grabsefx**
## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?
- Did you modify code to call some additional endpoints?
  - [ ] If so, do you check that necessary module permission added in ModuleDescriptor-template.yaml file?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/consortia/service/impl/UserAffiliationAsyncServiceImpl.java [hunk 1]
```diff
@@ -38,7 +38,7 @@ public CompletableFuture<Void> createPrimaryUserAffiliationsAsync(UUID consortiu
     Tenant tenantDto) {
     return CompletableFuture.runAsync(() -> {
         log.info("Start creating user primary affiliation for tenant {}", tenantDto.getId());
-        var users = userService.getUsersByQuery(StringUtils.EMPTY, 0, Integer.MAX_VALUE);
+        var users = userService.getUsersByQuery("cql.allRecords=1", 0, Integer.MAX_VALUE);
```
Comment (NikitaSedyx) [inline src/main/java/org/folio/consortia/service/impl/UserAffiliationAsyncServiceImpl.java:41]: it didn't help, error is the same

Comment (NikitaSedyx) [reply src/main/java/org/folio/consortia/service/impl/UserAffiliationAsyncServiceImpl.java:41]: error comes from https://github.com/folio-org/folio-spring-support/blob/master/folio-spring-base/src/main/java/org/folio/spring/client/EnrichUrlAndHeadersClient.java#L39 
smth connected to context

Comment (azizbekxm) [reply src/main/java/org/folio/consortia/service/impl/UserAffiliationAsyncServiceImpl.java:41]: I think CompletableFuture don't pass folioExecutionContext
image

File: src/main/java/org/folio/consortia/controller/TenantController.java [hunk 2]
```diff
@@ -43,4 +47,10 @@ public ResponseEntity<Void> deleteTenantById(UUID consortiumId, String tenantId)
     service.delete(consortiumId, tenantId);
     return ResponseEntity.status(NO_CONTENT).build();
   }
+
+  public ResponseEntity<Void> syncPrimaryAffiliations(UUID consortiumId, String tenantId,
+      SyncPrimaryAffiliationBody syncPrimaryAffiliationBody) {
+    userAffiliationAsyncService.createPrimaryUserAffiliationsAsync(consortiumId, syncPrimaryAffiliationBody);
```
Comment (NikitaSedyx) [inline src/main/java/org/folio/consortia/controller/TenantController.java:53]: this endpoint doesn't accept any body, only params consortiumId and tenantId

File: src/main/java/org/folio/consortia/service/impl/TenantServiceImpl.java [hunk 3]
```diff
@@ -116,7 +114,8 @@ public Tenant save(UUID consortiumId, UUID adminUserId, Tenant tenantDto) {
     // switch to context of the desired tenant and apply all necessary setup
     try (var context = new FolioExecutionContextSetter(contextHelper.getSystemUserFolioExecutionContext(tenantDto.getId()))) {
       configurationClient.saveConfiguration(createConsortiaConfigurationBody(centralTenantId));
-      createPrimaryUserAffiliationsAsync.createPrimaryUserAffiliationsAsync(consortiumId, savedTenantEntity, tenantDto);
+      // TODO: retrieve users list and POST sync-primary-affiliations endpoint
```
Comment (NikitaSedyx) [inline src/main/java/org/folio/consortia/service/impl/TenantServiceImpl.java:117]: `sync-primary-affiliations` should retrieve users

File: src/main/java/org/folio/consortia/service/impl/TenantServiceImpl.java [hunk 4]
```diff
@@ -135,9 +135,9 @@ public Tenant update(UUID consortiumId, String tenantId, Tenant tenantDto, Boole
     var tenantEntity = saveTenantEntity(consortiumId, tenantDto);
     var savedTenant = converter.convert(tenantEntity, Tenant.class);
 
-    if (forceCreatePrimaryAff.booleanValue()) {
+    if (forceCreatePrimaryAff) { //NOSONAR
```
Comment (NikitaSedyx) [inline src/main/java/org/folio/consortia/service/impl/TenantServiceImpl.java:138]: this is not needed

**Pull request closed without merging**